datagenerator/src/main/java/com/linkedin/restli/datagenerator/csharp/CSharpArray.java

```diff
@@ -35,11 +35,11 @@ public CSharpArray(ClassTemplateSpec spec, CSharpDataTemplateGenerator dataTempl
   }
 
   /**
-   * If param modifier is MUTABLE:
+   * If param modifier is MUTABLE_SHALLOW:
    *  Returns a mutable list of immutable types. The reason for this is that
    *  C# cannot handle casting a nested list/dictionary object; it can only handle
    *  casting the outer layer. Thus, the inner types must be immutable.
-   * If param modifier is DATAMAP_PARSE:
+   * If param modifier is TYPED_DATAMAP:
    *  Returns a mutable list of types that are readable by the constructors
    *  of each datamodel. For instance, all record types would be represented
    *  as Dictionary(string, object)
```

Comment:
Dictionary<string, object>. I thought that this is GENERIC_DATAMAP? Since we have commented clearly each NameModifier option in CSharpType, maybe we don't need to put these comments anymore.